editor/editor_help_search.cpp

```diff
@@ -394,6 +419,226 @@ bool EditorHelpSearch::Runner::_is_class_disabled_by_feature_profile(const Strin
 	return false;
 }
 
+bool EditorHelpSearch::Runner::_fill() {
+	bool phase_done = false;
+	switch (phase) {
+		case PHASE_MATCH_CLASSES_INIT:
+			phase_done = _phase_fill_classes_init();
+			break;
+		case PHASE_MATCH_CLASSES:
+			phase_done = _phase_fill_classes();
+			break;
+		case PHASE_CLASS_ITEMS_INIT:
+			phase_done = _phase_fill_member_items_init();
+			break;
+		case PHASE_CLASS_ITEMS:
+			phase_done = _phase_fill_member_items();
+			break;
+		case PHASE_MEMBER_ITEMS_INIT:
+		case PHASE_MEMBER_ITEMS:
+		case PHASE_SELECT_MATCH:
+		case PHASE_MAX:
+			return true;
+		default:
+			WARN_PRINT("Invalid or unhandled phase in EditorHelpSearch::Runner, aborting search.");
+			return true;
+	}
+
+	if (phase_done) {
+		phase++;
+	}
+	return false;
+}
+
+bool EditorHelpSearch::Runner::_phase_fill_classes_init() {
+	// Initialize fill.
+	iterator_stack.clear();
+	matched_classes.clear();
+
+	// Initialize stack of iterators to fill, in reverse.
+	iterator_stack.push_back(EditorHelp::get_doc_data()->inheriting[""].back());
+
+	return true;
+}
+
+bool EditorHelpSearch::Runner::_phase_fill_classes() {
+	if (iterator_stack.is_empty()) {
+		return true;
+	}
+
+	if (iterator_stack[iterator_stack.size() - 1]) {
+		DocData::ClassDoc *class_doc = EditorHelp::get_doc_data()->class_list.getptr(iterator_stack[iterator_stack.size() - 1]->get());
+
+		iterator_stack[iterator_stack.size() - 1] = iterator_stack[iterator_stack.size() - 1]->prev();
+		if (!iterator_stack[iterator_stack.size() - 1]) {
```

Comment:
Ditto?
Also repeating `size - 1` every time is a bit of a hassle, but I understand why it is done.

Reply:
It indeed has to be done as it isn't constant